src/microservices/Microservices.CohortPackager/Execution/ExtractJobStorage/MongoDB/ObjectModel/MongoExtractJobDoc.cs

```diff
@@ -37,6 +37,9 @@ public class MongoExtractJobDoc : MemberwiseEquatable<MongoExtractJobDoc>
         [BsonElement("keyCount")]
         public uint KeyCount { get; set; }
 
+        [BsonElement("userName")]
+        public string UserName { get; set; }
```

Comment:
We need to think about what happens here when we try and load an existing document from MongoDB which doesn't have this field. (I.e. a document created from an older version of code and stored in MongoDB without the UserName field). Can we write a test for this? Something like here: https://github.com/SMI/SmiServices/blob/5ecd5588f756eff3c31a7a238e2c090145a3b508/tests/microservices/Microservices.CohortPackager.Tests/Execution/ExtractJobStorage/MongoDB/ObjectModel/MongoFileStatusDocTest.cs#L98 ?

Reply:
Added test [here](https://github.com/SMI/SmiServices/blob/feature/1443-add-username-to-extraction-metadata/tests/microservices/Microservices.CohortPackager.Tests/Execution/ExtractJobStorage/MongoDB/ObjectModel/MongoExtractJobDocTest.cs#L118). It passes without any modifications to `MongoExtractJobDoc`, which surprised me because it's supposed to throw an exception if `UserName` is null in the constructor?

Reply:
Interesting! I don't think the constructor is being called by the deserializer, so the validation logic is being bypassed (https://mongodb.github.io/mongo-csharp-driver/1.11/serialization/#a-member-has-been-added). For this PR I think the test you added should be enough then.